### PR TITLE
AF-593: Migrate Data Modeller to AppFormer

### PR DESCRIPTION
**Pull request by mbarkley**

### CODE REVIEW

**Comment by mbarkley** [comment]
Related:
https://github.com/kiegroup/guvnor/pull/519
https://github.com/dashbuilder/dashbuilder/pull/372
https://github.com/kiegroup/kie-soup/pull/3
https://github.com/kiegroup/kie-wb-distributions/pull/621
https://github.com/kiegroup/droolsjbpm-tools/pull/82
https://github.com/kiegroup/optaplanner-wb/pull/218
https://github.com/kiegroup/jbpm-wb/pull/864
https://github.com/kiegroup/jbpm-designer/pull/654
https://github.com/kiegroup/drools-wb/pull/596
https://github.com/kiegroup/kie-wb-common/pull/1133
https://github.com/kiegroup/droolsjbpm-integration/pull/1163
https://github.com/kiegroup/droolsjbpm-integration/pull/1163
https://github.com/kiegroup/jbpm/pull/973
https://github.com/kiegroup/drools/pull/1444
https://github.com/kiegroup/droolsjbpm-knowledge/pull/263
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/541
https://github.com/AppFormer/uberfire/pull/822

**Comment by mbarkley** [comment]
@psiroky The build fails because it cannot construct a repository list. Would you mind taking a look?

**Comment by psiroky** [comment]
@mbarkley works now. It failed because https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/541 was not rebased against latest changes on master (at the time of this PR build).

**Comment by mbarkley** [comment]
@porcelli @mariofusco @manstis This PR has the project-datamodel that I had previously relocated to UberFire. It was also necessary to move some types from uberfire-commons here (hence the new kie-soup-commons).

**Comment by mbarkley** [comment]
@porcelli @manstis @ederign Anyone able to review this? This is the last PR in the set that needs review.

File: kie-soup-commons/pom.xml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2015 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline kie-soup-commons/pom.xml]: 2017

File: kie-soup-commons/src/main/java/org/kie/soup/commons/validation/PortablePreconditions.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline kie-soup-commons/src/main/java/org/kie/soup/commons/validation/PortablePreconditions.java]: 2017

File: kie-soup-project-datamodel/kie-soup-project-datamodel-api/.classpath [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (manstis) [inline kie-soup-project-datamodel/kie-soup-project-datamodel-api/.classpath]: Don't need this file. Do you have a ```.gitignore``` in this PR?

File: kie-soup-project-datamodel/kie-soup-project-datamodel-commons/.classpath [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (manstis) [inline kie-soup-project-datamodel/kie-soup-project-datamodel-commons/.classpath]: Please remove.

**Comment by mbarkley** [comment]
@porcelli @manstis Done.

File: kie-soup-project-datamodel/kie-soup-project-datamodel-commons/src/main/java/org/kie/soup/project/datamodel/commons/util/RawMVELEvaluator.java [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+ */
+package org.kie.soup.project.datamodel.commons.util;
+
+import java.util.Map;
+
+import org.mvel2.MVEL;
+import org.mvel2.compiler.CompiledExpression;
+import org.mvel2.integration.VariableResolverFactory;
+
+public class RawMVELEvaluator implements MVELEvaluator {
```
Comment (manstis) [inline kie-soup-project-datamodel/kie-soup-project-datamodel-commons/src/main/java/org/kie/soup/project/datamodel/commons/util/RawMVELEvaluator.java]: @mbarkley Is this meant to be the default implementation?

I'm unable to run ```drools-wb-webapp``` as there are no CDI Beans satisfying injection points for ```MVELEvaluator``` in ```org.kie.workbench.common.services.backend.enums.EnumDropdownServiceImpl```. I can make this implementation ```@ApplicationScoped``` and add a dependency for ```javax.enterprise:cdi-api``` however IDK if ```kie-soup``` is meant to be CDI-free and we need a _producer_ method in kie-wb-common/guvnor instead!?

File: .gitignore [hunk 6]
```diff
@@ -23,3 +23,8 @@ hs_err_pid*
 
 # Maven build directories
 **/target/
+
+#Eclipse Files
```
Comment (manstis) [inline .gitignore]: Any chance you can add the defaults for IDEA and Netbeans(!!) too?

https://github.com/kiegroup/kie-wb-common/blob/master/.gitignore#L10

File: kie-soup-project-datamodel/kie-soup-project-datamodel-commons/src/main/java/org/kie/soup/project/datamodel/commons/util/RawMVELEvaluator.java (hunk 5 above)
Comment (manstis) [reply kie-soup-project-datamodel/kie-soup-project-datamodel-commons/src/main/java/org/kie/soup/project/datamodel/commons/util/RawMVELEvaluator.java]: OK... I needed ```drools-cdi``` dependency too. All fixed.

File: .gitignore (hunk 6 above)
Comment (mbarkley) [reply .gitignore]: @manstis Fixed [here](https://github.com/kiegroup/kie-soup/pull/5).

**Pull request merged**